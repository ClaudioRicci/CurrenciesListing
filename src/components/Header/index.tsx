import React, { memo } from "react";
import shortid from "shortid";
import "./styles/Header.scss";

interface HeaderProps {
  title: string;
}

const Header = ({ title }: HeaderProps) => (
  <caption key={shortid.generate()}>{title}</caption>
);

export default memo(Header);
