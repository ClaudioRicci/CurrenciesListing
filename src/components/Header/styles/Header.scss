caption {
  color: #00008b;
  margin: 2rem auto;
  text-align: center;
  font-weight: 900;
}
