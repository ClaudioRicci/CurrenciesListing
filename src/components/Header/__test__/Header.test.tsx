import React from "react";
import renderer from "react-test-renderer";
import { shallow } from "enzyme";
import Enzyme from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import { render, cleanup } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
Enzyme.configure({ adapter: new Adapter() });
import Header from "..";

afterEach(cleanup);

describe("Header Tag Tests", () => {
  test("Renders Header", () => {
    const { getByText } = render(<Header title="Currencies Table" />);
    const headerCopy = getByText(/Currencies Table/i);
    expect(headerCopy).toBeInTheDocument();
  });

  test("Caption Renders Correctly and matches Snapshot", () => {
    const tree = renderer.create(<Header title="test" />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
