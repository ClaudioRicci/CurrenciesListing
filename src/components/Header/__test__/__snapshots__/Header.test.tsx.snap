// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Tag Tests Caption Renders Correctly and matches Snapshot 1`] = `
<caption>
  test
</caption>
`;
